src/vdom.rs

```diff
@@ -72,45 +103,50 @@ pub(crate) fn build(host: &Host, input: VDom<()>) -> VDomNode<Vec<VDom<Node>>, N
                 })
                 .collect();
             // Return the machine
-            VDomNode::Elem {
-                name,
-                attrs: attrs_new,
-                children,
-                state: node,
-            }
+            VDomNode::new_elem(elem.name, attrs_new, children, node)
         }
     }
 }
 
 impl VDomNode<Vec<VDom<Node>>, Node, Attrs> {
+    pub fn new_text(text: String, state: Node) -> Self {
```

Comment:
new_text and new_elem don't need to be specific to [Node], they can be defined for any [State]